File: TimelineSorter.js
class TimelineSorter {
    sort(events) {
        const validEvents = [];
        const undatedEvents = [];

        events.forEach(event => {
            if (event.timestamp && !isNaN(new Date(event.timestamp).getTime())) {
                validEvents.push(event);
            } else {
                undatedEvents.push(event);
            }
        });

        validEvents.sort((a, b) => {
            return new Date(a.timestamp) - new Date(b.timestamp);
        });

        return [...validEvents, ...undatedEvents];
    }
}

if (typeof window === 'undefined') {
    const sorter = new TimelineSorter();
    const data = [
        { id: 1, timestamp: "2023-12-25 08:00:00", name: "Christmas" },
        { id: 2, name: "Old photo with no date" },
        { id: 3, timestamp: "2023-01-01 00:00:00", name: "New Year" },
        { id: 4, timestamp: "invalid-date", name: "Error entry" }
    ];

    console.log(JSON.stringify(sorter.sort(data), null, 2));
}

export default TimelineSorter;

File: timeline_sorter.py
import json
from datetime import datetime

class TimelineSorter:
    def __init__(self):
        self.fmt = "%Y-%m-%d %H:%M:%S"

    def parse(self, date_str):
        try:
            return datetime.strptime(date_str, self.fmt)
        except:
            return None

    def sort_items(self, items):
        valid = []
        no_date = []

        for item in items:
            if 'timestamp' in item and self.parse(item['timestamp']):
                valid.append(item)
            else:
                no_date.append(item)

        valid.sort(key=lambda x: self.parse(x['timestamp']))
        
        return valid + no_date

if __name__ == "__main__":
    sorter = TimelineSorter()
    
    data = [
        {"id": 1, "timestamp": "2023-12-25 08:00:00", "name": "Christmas"},
        {"id": 2, "name": "Old photo with no date"},
        {"id": 3, "timestamp": "2023-01-01 00:00:00", "name": "New Year"},
        {"id": 4, "timestamp": "broken-date", "name": "Error entry"}
    ]

    result = sorter.sort_items(data)
    print(json.dumps(result, indent=4))